### ISSUE TITLE
Could not find configuration node: reporting/dataProvider on CD

### ISSUE THREAD

**Issue by estebanCapello** [WeTeam/WeBlog#240]
After upgrading weblog for sitecore8.2 into a CD - CM environment ( same network ). For the CD server  I activated the weBlog.delivery.config and deactivated the following files:
- WeBlog.config
- WeBlog.ContentSearch.Master.config

But I'm when trying to access into my main blog display page i get the following:

`Could not find configuration node: reporting/dataProvider`
`[InvalidOperationException: Could not find configuration node: reporting/dataProvider]
   Sitecore.Diagnostics.Assert.IsTrue(Boolean condition, String message) +57
   Sitecore.Configuration.DefaultFactory.GetConfigNode(String xpath, Boolean assert) +298
   Sitecore.Configuration.DefaultFactory.CreateObject(String configPath, String[] parameters, Boolean assert) +284
   Sitecore.Analytics.Reporting.ReportingQueryBase..ctor(ReportDataProviderBase reportProvider, CachingPolicy cachingPolicy) +40
   Sitecore.Modules.WeBlog.Managers.EntryManager.GetItemViews(ID itemId) +77
   Sitecore.Modules.WeBlog.Managers.EntryManager.GetPopularEntriesByView(Item blogItem, Int32 maxCount) +358
   Sitecore.Modules.WeBlog.WebForms.Layouts.Sidebar.BlogInterestingEntries.Page_Load(Object sender, EventArgs e) +59
   Sitecore.Modules.WeBlog.WebForms.Layouts.BaseSublayout.OnLoad(EventArgs e) +90
   System.Web.UI.Control.LoadRecursive() +71
   System.Web.UI.Control.LoadRecursive() +190
   System.Web.UI.Control.LoadRecursive() +190
   System.Web.UI.Control.LoadRecursive() +190
   System.Web.UI.Control.LoadRecursive() +190
   System.Web.UI.Control.LoadRecursive() +190
   System.Web.UI.Control.LoadRecursive() +190
   System.Web.UI.Control.LoadRecursive() +190
   System.Web.UI.Page.ProcessRequestMain(Boolean includeStagesBeforeAsyncPoint, Boolean includeStagesAfterAsyncPoint) +3178`

Is this a bug or I'm missing additional configurations for a CD server ?

**Comment by alan-null** [comment]
Hello @estebanCapello,

Where did you find such information?
The package readme file says:
>If deploying in a staged environment to a content delivery
server, enable the App_Config\Include\WeBlog.delivery.config
file.


The error says that it cannot find following configuration node `reporting/dataProvider`.
Could you validate that you can find following config file `Sitecore.Analytics.Reporting.config` or check merged configuration
![image](https://cloud.githubusercontent.com/assets/6848691/24035584/7783b79a-0af6-11e7-98a0-a715b425c44b.png)

**Comment by alan-null** [comment]
**TLDR:** Make sure that your server has direct access to the reporting server or issues queries against a remote reporting service:

------

Ok, let's get back to issue with rendering **InterestingEntries** component.

This is a component which uses reporting db to fetch analytics data.
After you disabled `Sitecore.Analytics.Reporting.config` config file (according to Sitecore CD recommendation doc) you will not be able to render a component as **InterestingEntries** relies on analytics.



There are two solutions:
* Assuming you are not interested in using **InterestingEntries** component you could just remove it from presentation details and WeBlog should work fine both on CD and CM
* There is a possible solution if you want to use **InterestingEntries** but I haven't tested it. You will have to provide connection to a **Reporting Service server** ([configure_a_reporting_service_server](https://doc.sitecore.net/sitecore_experience_platform/setting_up_and_maintaining/xdb/configuring_servers/configure_a_reporting_service_server)). so **InterestingEntries** can consume necessary data and then  render.

**Comment by estebanCapello** [comment]
Thanks a lot for the quick help !!!

At the end I followed your first suggestion. Currently I'm not using InterestingEntries so I deleted it from the presentation details and it worked !
